### PR TITLE
improve performance : excessive check removed

### PR DESCRIPTION
**Pull request by kptfh**

### CODE REVIEW

File: spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/SleuthAdvisorConfig.java [hunk 1]
```diff
@@ -105,7 +105,7 @@ private Pointcut buildPointcut() {
 
 		@Override
 		public boolean matches(Method method, Class<?> targetClass, Object... args) {
-			return getClassFilter().matches(targetClass);
```
Comment (marcingrzejszczak) [inline spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/SleuthAdvisorConfig.java]: Doesn't it break any logic?

Comment (kptfh) [reply spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/SleuthAdvisorConfig.java]: No. Actual check takes place in SleuthInterceptor.invoke(MethodInvocation)

Comment (gavlyukovskiy) [reply spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/SleuthAdvisorConfig.java]: Doesn't it mean that all classes will be proxied regardless of existence of `NewSpan` or `ContinueSpan` annotations?

Comment (marcingrzejszczak) [reply spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/SleuthAdvisorConfig.java]: Yeah, that's what i was afraid of. But after looking at the code very briefly and following @kptfh suggestions, it seems that the advisor will apply the interceptor if there's an annotation on a class or a method. But maybe I'm wrong, like I said i haven't read the code thoroughly.

Comment (kptfh) [reply spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/SleuthAdvisorConfig.java]: No. Check for class remains unchanged `getClassFilter()`. Just remove duplicated check for method being annotated.

**Pull request merged**